File: samples/python/03sqrt/sqrt.py
import sys

def find_sqrt(number):
    number = float(number)
    x = number
    y = 1.0
    e = 0.000000001
    while x - y > e:
        x = (x + y) / 2.0
        y = number / x
    return x


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Error: Usage python sqrt.py <number> [<number> ...]")
        sys.exit(1)
    
    for arg in sys.argv[1:]:
        try:
            number = float(arg)
        except ValueError:
            print("Error: Expected a number. Received", arg)
            sys.exit(2)
        print("%f => %f" % (number, find_sqrt(number)))

File: samples/ruby/03sqrt/sqrt.rb
ACCURACY = 0.0000001

def findSqrt(number)
    x = number
    y = 1.0
    while x-y > ACCURACY
        x = (x + y) / 2.0
        y = number / x
    end
    return x
end

if ARGV.length < 1
    $stderr.puts "Error: Usage `ruby sqrt.rb <number> [<number> ...]`"
    exit(1)
end

for arg in ARGV
    if (false if Float(arg) rescue true)
        $stderr.puts "Error: Expected to receive a number. Received '%s'" % [arg]
        exit(2)
    end

    number = arg.to_f

    puts "%f => %f" % [number, findSqrt(number)]
end
